test/jest/unit/snyk-code/snyk-code-test.spec.ts

```
@@ -380,6 +386,55 @@ describe('Test snyk code', () => {
     }
   });
 
+  it('should create sarif result with ignored issues omitted', async () => {
+    const sastSettings = {
+      sastEnabled: true,
+      localCodeEngine: { url: '', allowCloudUpload: true, enabled: false },
+    };
+
+    // First get results without ignores
+    analyzeFoldersMock.mockResolvedValue(sampleAnalyzeFoldersResponse);
+    const resultWithoutIgnores = await getCodeTestResults(
+      '.',
+      {
+        path: '',
+        code: true,
+      },
+      sastSettings,
+      'test-id',
+    );
+
+    const sarifWithoutIgnores =
+      resultWithoutIgnores?.analysisResults.sarif.runs[0].results;
+    if (!sarifWithoutIgnores) return;
```

enh: We should really, really be failing the test if this happens. Silently early-exiting tests is a terrible practice and a good way to get silently failing tests. See also below.